src/content/docs/en/tutorial/add-content-collections.mdx

```diff
@@ -0,0 +1,327 @@
+      pnpm install @astrojs/preact@latest
+      ```
+      </Fragment>
+      <Fragment slot="yarn">
+      ```shell
+      # Upgrade to Astro v3.x
+      yarn add astro@latest
+
+    # Example: upgrade the blog tutorial Preact integration
+      yarn add @astrojs/preact@latest
+      ```
+      </Fragment>
+    </PackageManagerTabs>
+
+    :::tip
+    If you are using your own project, then be sure to update any dependencies you have installed. The example blog tutorial codebase only uses the Preact integration.
+    :::
+
+2. The blog tutorial uses the `base` (least strict) TypeScript setting. In order to use content collections, you must [set up TypeScript](/en/guides/content-collections/#setting-up-typescript) for content collections either by using the `strict` or `strictest` setting, or by adding two options in `tsconfig.json`.
+
+    In order to use content collections without writing TypeScript in the rest of the blog tutorial example, add the following two TypeScript configuration options:
+
+    ```json title="tsconfig.json" ins={5,6}
+    {
+      // Note: No change needed if you use "astro/tsconfigs/strict" or "astro/tsconfigs/strictest"
+      "extends": "astro/tsconfigs/base",
+      "compilerOptions": {
+        "strictNullChecks": true,
+        "allowJs": true
+      }
+    }
+    ```
+
+### Create a collection for your blog posts
+
+3. Create a new **collection** (folder) called `src/content/posts/`. 
+
+4. Move all your existing blog posts (`.md` files) from `src/pages/posts/` into this new collection.
+
+5. Create a `src/content/config.ts` file to [define a schema](/en/guides/content-collections/#defining-a-collection-schema) for your `postsCollection`:
+
+    ```ts title="src/content/config.ts"
+    // Import utilities from `astro:content`
+    import { z, defineCollection } from "astro:content";
+    // Define a `type` and `schema` for each collection
+    const postsCollection = defineCollection({
+        type: 'content',
+        schema: z.object({
+          title: z.string(),
+          pubDate: z.date(),
+          description: z.string(),
+          author: z.string(),
+          image: z.object({
+            url: z.string(),
+            alt: z.string()
+          }),
+          tags: z.array(z.string())
+        })
+    });
+    // Export a single `collections` object to register your collection(s)
+    export const collections = {
+      posts: postsCollection,
+    };
+    ```
+
+### Generate pages from a collection
+
+6. Create a page file called `src/pages/posts/[...slug].astro`. Your Markdown and MDX files no longer automatically become pages using Astro's file-based routing when they are inside a collection, so you must create a page responsible for generating each individual blog post.
+
+7. Add the following code to [query your collection](#querying-collections) to make each blog post's slug and page content available to each page it will generate:
```

Comment:
```suggestion
7. Add the following code to [query your collection](/en/guides/content-collections/#querying-collections) to make each blog post's slug and page content available to each page it will generate:
```